app/pages/common/dashboard/controller.test.js

```diff
@@ -1,75 +1,178 @@
-            id: 'some-section-group-id',
-            title: 'Some section group',
-            sections: [
-              {
-                URL: 'some-solution-id/section/some-section-id',
-                id: 'some-section-id',
-                title: 'Some section',
-                requirement: 'Mandatory',
-                status: 'INCOMPLETE',
-                isActive: true,
-              },
-            ],
+    },
+  },
+};
+
+const dashboardData = {
+  name: 'some solution name',
+  'supplier-name': 'a supplier',
+  sections: {
+    'some-section-id': {
+      status: 'INCOMPLETE',
+      requirement: 'Mandatory',
+    },
+  },
+};
+
+describe('supplier - dashboard controller', () => {
+  describe('getMarketingPageDashboardContext', () => {
+    describe('userContextType not specified', () => {
+      afterEach(() => {
+        manifestProvider.getDashboardManifest.mockReset();
+        apiProvider.getData.mockReset();
+        context.createDashboardPageContext.mockReset();
+      });
+
+      it('should call getDashboardManifest with the correct params', async () => {
+        manifestProvider.getDashboardManifest.mockReturnValueOnce(dashboardManifest);
+        apiProvider.getData.mockResolvedValueOnce(dashboardData);
+
+        await getMarketingPageDashboardContext({ solutionId: 'some-solution-id', dashboardId: 'some-dashboard-id' });
+
+        expect(manifestProvider.getDashboardManifest.mock.calls.length).toEqual(1);
+        expect(manifestProvider.getDashboardManifest).toHaveBeenCalledWith({ userContextType: 'supplier' });
+      });
+
+      it('should call getData with the correct params', async () => {
+        apiProvider.getData.mockResolvedValueOnce(dashboardData);
+
+        await getMarketingPageDashboardContext({ solutionId: 'some-solution-id', dashboardId: 'some-dashboard-id' });
+
+        expect(apiProvider.getData.mock.calls.length).toEqual(1);
+        expect(apiProvider.getData).toHaveBeenCalledWith({
+          endpointLocator: 'getMainDashboardData',
+          options: {
+            solutionId: 'some-solution-id',
+            userContextType: 'supplier',
           },
-        ],
-      };
+        });
+      });
+
+      it('should call createDashboardPageContext with the correct params', async () => {
+        manifestProvider.getDashboardManifest.mockReturnValueOnce(dashboardManifest);
+        apiProvider.getData.mockResolvedValueOnce(dashboardData);
+
+        await getMarketingPageDashboardContext({ solutionId: 'some-solution-id', dashboardId: 'some-dashboard-id' });
+
+        expect(context.createDashboardPageContext.mock.calls.length).toEqual(1);
+        expect(context.createDashboardPageContext).toHaveBeenCalledWith({
+          solutionId: 'some-solution-id',
+          solutionName: dashboardData.name,
+          supplierName: dashboardData['supplier-name'],
+          dashboardManifest,
+          marketingDataSections: dashboardData.sections,
+          validationErrors: undefined,
```

Comment:
can remove this from the check

Reply:
no I can't as it checks the whole object and this is what is passed through if there are no validation errors

Reply:
ok